onitu/referee/folder.py

```diff
@@ -31,82 +18,115 @@ def __repr__(self):
 
-        if not self.check_size(metadata['size']):
-            self.logger.info(
-                "Ignoring event for '{}' in folder {} due to its size: "
-                "{} bytes",
-                filename, self.name, metadata['size']
-            )
-            return
+        # To decide which targets are relevant there are two things we
+        # should take into account. folder options and service/folder
+        # options.
 
-        if not self.check_mimetype(metadata['mimetype']):
-            self.logger.info(
-                "Ignoring event for '{}' in folder {} due to its mimetype: {}",
-                filename, self.name, metadata['mimetype']
-            )
-            return
+        # TODO: We might want to consider doing different checks for the
+        # source than what we do for the targets because in some cases
+        # we might want to cheat. Eg: what happens if the size grows to
+        # large for the source's service, we'dd still like for it to be
+        # copied. Or not? Maybe not. I don't know. Not doing it now.
 
-        if self.blacklisted(filename):
-            self.logger.info(
-                "Ignoring event for '{}' in folder {} because its filename "
-                "is blacklisted",
-                filename, self.name
-            )
-            return
+        # Step 1: Do we (the folder) want the file?
 
-        if not self.whitelisted(filename):
-            self.logger.info(
-                "Ignoring event for '{}' in folder {} because its filename "
-                "is not whitelisted",
-                filename, self.name
-            )
-            return
+        if not self.assert_options(self.options, metadata):
+            return set()
 
-        return self.services - frozenset((source,))
+        # Step 2: Does the source want to share it?
 
-    def check_size(self, size):
-        if self.min_size is not None and size < self.min_size:
-            return False
+        options = self.services[source]
+        if not self.assert_options(options, metadata):
+            return set()
+
+        # Step 3: Check who else is interested.
+
+        targets = set()
+        for service, options in self.services.items():
+            if service == source:
+                continue
+            if self.assert_options(options, metadata):
+                targets.add(service)
+
+        return targets
+
+    def assert_options(self, options, metadata, mode=""):
+
+        # Those options are either our own or those of a service.
+        # This is because there is no difference between
+        # service/folder and folder options.
+
+        def check_list(l, c):
+            if l is None:
+                return None
+            return any(fnmatchcase(c, rule) for rule in l)
 
-        if self.max_size is not None and size > self.max_size:
+        # mode
+
+        modeopts = options.get("mode", "rw")
+        if any(c not in modeopts for c in mode):
```

Comment:
`mode` is an argument to the `assert_options` method telling it if we are checking in order to read `r` or write `w` the file. These particular lines check if any character in mode isn't in the allowed mode specified in the options.
This way we are consistent regarding the syntax for modes everywhere and one could pass `rw` as argument to the `assert_options` method without problems if this is needed some time.